### ISSUE TITLE
Coder/code-server conflicts with the startup command of visual studio code server[Bug]: 

### ISSUE THREAD

**Issue by 2659170494** [coder/code-server#5502]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS/Web Information

- Web Browser:Chrome
- Local OS:Windows
- Remote OS:Ubuntu
- Remote Architecture:amd64
- `code-server --version`:4.4.0

### Steps to Reproduce

1To install the coder/code-server (or visual studio code server)
2.run "code-server" then you will see the estimated results
3.instal the visual studio code server (or coder/code-server)
4. run "code-server" then you will find that the results are different from the original ones

### Expected

This will start code server before installation

### Actual

After installing visual studio code server, the startup command of visual studio code server overwrites the original startup command. However, the original code server startup sh file can still be found by searching

### Logs

No has log

### Does this issue happen in VS Code or GitHub Codespaces?

- [X] I cannot reproduce this in VS Code.
- [X] I cannot reproduce this in GitHub Codespaces.

### Are you accessing code-server over HTTPS?

- [X] I am using HTTPS.

### Notes

The code server file of visual studio code server is located in /usr/local/bin/, and this file may be a binary file

At present, visual studio code server is still under internal test, so few people may have this bug before it is officially launched. I want to know if code server has any startup commands other than code server?

**Comment by jsjoeio** [comment]
> After installing visual studio code server, the startup command of visual studio code server overwrites the original startup command.

I feel like this bug should be filed with them, don't you think? That shouldn't be overwriting our startup command. They should check for conflicts and handle them if they exist. Have you tried contacting them about it?

**Comment by 2659170494** [comment]
> > After installing visual studio code server, the startup command of visual studio code server overwrites the original startup command.
> 
> I feel like this bug should be filed with them, don't you think? That shouldn't be overwriting our startup command. They should check for conflicts and handle them if they exist. Have you tried contacting them about it?

Yes, I just got the qualification of internal test. After the problem occurs, I put forward the issue in both the internal test warehouse and this warehouse. However, what they got was that because the warehouse was not Microsoft, they could not solve the problem and closed the issue

**Comment by benz0li** [comment]
@jsjoeio I think @2659170494 is talking about https://github.com/coder/code-server/issues/5417#issuecomment-1210203300, i.e. executing `code-server` in code-server Terminal not being the same as executing `code-server` in a [regular] JupyterLab Terminal (e.g. https://demo.jupyter.b-data.ch).

**Comment by benz0li** [comment]
> The only thing theirs [Microsoft's] does not support is launching a new instance of code-server (it only supports interacting with the current instance of code-server) but I have no idea if that is a common use in the integrated terminal.

_Originally posted by @code-asher in https://github.com/coder/code-server/issues/5360#issuecomment-1191866815_

**Comment by jsjoeio** [comment]
> executing code-server in code-server Terminal not being the same as executing code-server in a [regular] JupyterLab Terminal

Ohhh...Whoops! Sorry for the confusion. Thanks for pointing that out @benz0li.  @2659170494 does that sound right? You're trying to execute `code-server` in a JupyterLab Terminal?

**Comment by benz0li** [comment]
> You're trying to execute `code-server` in a JupyterLab Terminal?

@2659170494 Or any regular Terminal [other than code-server Terminal]?

**Comment by 2659170494** [comment]
> > You're trying to execute  in a JupyterLab Terminal?`code-server`
> 
> @2659170494 Or any regular Terminal [other than code-server Terminal]?



> > executing code-server in code-server Terminal not being the same as executing code-server in a [regular] JupyterLab Terminal
> 
> Ohhh...Whoops! Sorry for the confusion. Thanks for pointing that out @benz0li. @2659170494 does that sound right? You're trying to execute in a JupyterLab Terminal?`code-server`

Maybe, I use SSH to control my server. I haven't used other terminal software, and my server is an ordinary Ubuntu system (but I started the jupyterlab terminal mentioned by benz0li according to the link he gave, but the server was started incorrectly 🤔)。 I mean the conflict of startup commands. When I run code server, the server starts visual studio code server instead of code server. Visual studio code server replaces the startup command of code server during installation

**Comment by benz0li** [comment]
> (but I started the jupyterlab terminal mentioned by benz0li according to the link he gave, but the server was started incorrectly 🤔)

If the server at https://demo.jupyter.b-data.ch does not start, simply give it another try.

**Comment by benz0li** [comment]
@2659170494 When you talk about _Visual Studio Code Server_ you mean https://code.visualstudio.com/docs/remote/vscode-server, correct?

**Comment by 2659170494** [comment]
> @2659170494当你谈论_Visual Studio Code Server_时，你的意思是 https://code.visualstudio.com/docs/remote/vscode-server，对吗？

Yes, I have only recently obtained the qualification for internal testing

**Comment by 2659170494** [comment]
> > （但是我根据Benz0li给出的链接启动了benz0li提到的jupyterlab终端，但是服务器启动不正确🤔)
> 
> 如果[处于 https://demo.jupyter.b-data.ch](https://demo.jupyter.b-data.ch) 的服务器无法启动，只需再试一次即可。

OK, I just started it successfully😂

**Comment by benz0li** [comment]
[Using `wget -O- https://aka.ms/install-vscode-server/setup.sh | sh`] Visual Studio Code Server installs at `/usr/local/bin`.

[Using `dpkg -i code-server_4.6.0_amd64.deb`] coder/code-server installs at `/usr/bin`.

```bash
$ cat /usr/bin/code-server
```

```sh
#!/usr/bin/env sh

exec /usr/lib/code-server/bin/code-server "$@"
```

ℹ️ I do not see any files getting overwritten.

But trying to start coder/code-server by executing `code-server` from within Visual Studio Code Server's *integrated Terminal* or vice versa will fail, because the [`PATH` gets overwritten](https://github.com/coder/code-server/issues/5417#issuecomment-1210203300).

---

I advise against installing Visual Studio Code Server and coder/code-server on the same machine.

**Comment by benz0li** [comment]
> When I run code server, the server starts visual studio code server instead of code server. Visual studio code server replaces the startup command of code server during installation

```bash
echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

**Comment by benz0li** [comment]
> When I run code server, the server starts visual studio code server instead of code server. Visual studio code server replaces the startup command of code server during installation

No. `/usr/local/bin` simply takes precedence over `/usr/bin`.

```bash
echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

**Comment by 2659170494** [comment]
> > When I run code server, the server starts visual studio code server instead of code server. Visual studio code server replaces the startup command of code server during installation
> 
> No. simply takes precedence over .`/usr/local/bin``/usr/bin`
> 
> ```shell
> echo $PATH
> /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
> ```

Yes, I know that. I can also start code server under the code folder of code server. However, I am more concerned about whether there are other startup commands for code server?

**Comment by code-asher** [comment]
I am not sure I followed the issue, but if you install code-server then you install something else on the PATH that also has a binary called `code-server`, then there is nothing we can do about that conflict.

> However, I am more concerned about whether there are other startup commands for code server?

`code-server` is the only command we have.

**Comment by 2659170494** [comment]
Ah, it looks like this is a very old issue.

I also raised the same issue in the repositories of VSCode remote release when this issue was released.
However, they also have no intention of dealing with this [issue](https://github.com/microsoft/vscode-remote-release/issues/7122)😂

Now when I roughly review the documents, it seems that they no longer use this command, or in other words, the problem has already been solved by them?
